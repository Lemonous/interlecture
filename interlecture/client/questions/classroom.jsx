import React from 'react';
import { ListGroup, ListGroupItem, Grid, Row, Col, Form } from 'react-bootstrap';
import 'static/css/font-awesome.css';
import { connect } from 'react-redux';
import PostList from './postList';
import InputForm from './inputForm';

function mapStateToProps(state) {
  return {
    posts: state.posts.filter(v => v),
    replyTo: state.replyTo,
  };
}

export const Classroom = ({ classroom, submitQuestion, ...props }) => (
  <div>
    <div style={{ margin: '50px 5% 40% 5%' }}>
      <PostList {...props} />
    </div>

    <div
      style={{
        position: 'fixed',
        bottom: '0px',
        width: '100%',
        height: '50px',
        paddingTop: '10px',
        backgroundColor: '#fff',
      }}
    >
      <div
        style={{
          width: '85%',
          margin: '0 auto',
        }}
      >
        <InputForm
          onSubmit={submitQuestion}
          id={'submitQuestion'}
          placeholder={'Enter question'}
          submitButtonText={'Submit Question'}
        />
      </div>
    </div>
  </div>
);

Classroom.propTypes = {
  classroom: React.PropTypes.shape({
    id: React.PropTypes.number,
    title: React.PropTypes.string,
    lecturer: React.PropTypes.string,
  }).isRequired,
  submitQuestion: React.PropTypes.func.isRequired,
};

export default connect(mapStateToProps)(Classroom);
