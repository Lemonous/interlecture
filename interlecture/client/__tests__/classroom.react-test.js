import React from 'react';
import renderer from 'react-test-renderer';
import { Classroom } from '../questions/classroom.jsx';

const questions = [
  {
    id: 1,
    user: 'Rolf',
    body: 'hvorfor gjør vi dette?',
    datetime: 'whatever',
  },
  {
    id: 2,
    user: 'Arne',
    body: 'kommer dette på eksamen?',
    datetime: 'whatever',
  },
  {
    id: 3,
    user: 'Kåre',
    body: 'hva er meningen med livet?',
    datetime: 'whatever',
  },
  {
    id: 4,
    user: 'Bjarne',
    body: 'finnes egentlig julenissen?',
    datetime: 'whatever',
  },
  {
    id: 5,
    user: 'Rolf',
    body: 'jeg skjønner ikke håndskriften din :(',
    datetime: 'whatever',
  },
];

function onSubmit(event) {
  event.preventdefault();
  console.log('onSubmit');
}

const classroom = {
  id: 0,
  title: 'test',
  lecturer: 'gruppe69',
};

test('Classroom looks like it should', () => {
  const component = renderer.create(
    <Classroom
      classroom={classroom}
      submitQuestion={onSubmit}
      submitReply={onSubmit}
      submitLike={onSubmit}
      questions={questions}
    />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
