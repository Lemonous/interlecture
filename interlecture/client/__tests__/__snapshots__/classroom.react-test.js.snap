// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Classroom looks like it should 1`] = `
<div
  style={
    Object {
      "marginBottom": "40px",
      "marginTop": "50px",
    }
  }
>
  <h2>
    test
  </h2>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-4"
      />
      <div
        className="col-sm-4"
      />
      <div
        className="col-sm-4"
      >
        <p>
          <b>
            lecturer: 
            gruppe69
          </b>
        </p>
      </div>
    </div>
  </div>
  <ul
    className="list-group"
  >
    <li
      className="list-group-item"
      style={
        Object {
          "border": "1px solid #888",
          "borderRadius": 5,
          "boxShadow": "0px 5px 35px rgba(0, 0, 0, .7)",
          "marginBottom": "30px",
          "marginLeft": "30px",
          "marginRight": "30px",
          "padding": "0px 0px",
          "paddingBottom": "5px",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#fff",
            "padding": "10px",
          }
        }
      >
        <div
          className="container"
        >
          <div
            className="show-grid row"
          >
            <div
              className="col-md-8 col-xs-8"
            >
              <div>
                <p>
                  <span
                    aria-hidden={true}
                    className="fa fa-user"
                  />
                   
                  <b>
                    Rolf
                  </b>
                </p>
                <p>
                     
                </p>
                <button
                  className="btn btn-default"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Like  
                  <span
                    aria-hidden={true}
                    className="fa fa-thumbs-up"
                  />
                </button>
                <button
                  className="btn btn-default"
                  disabled={false}
                  type="button"
                >
                  Reply  
                  <span
                    aria-hidden={true}
                    className="fa fa-reply"
                  />
                </button>
                <button
                  className="btn btn-default"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Delete  
                  <span
                    aria-hidden={true}
                    className="fa fa-remove-circle"
                  />
                </button>
              </div>
            </div>
            <div
              className="col-md-4 col-xs-4"
            >
              <p>
                <span
                  aria-hidden={true}
                  className="fa fa-clock-o"
                />
                 
                whatever
              </p>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "paddingLeft": "5px",
              "paddingRight": "5px",
              "width": "100%",
            }
          }
        >
          <button
            className="btn btn-primary"
            form="replyTo1"
            style={
              Object {
                "float": "right",
              }
            }
            type="submit"
            value="submitButton"
          >
            Submit Reply
          </button>
          <div
            style={
              Object {
                "overflow": "hidden",
                "paddingRight": "0.5rem",
              }
            }
          >
            <form
              className=""
              id="replyTo1"
              onSubmit={[Function]}
            >
              <input
                className="form-control"
                placeholder="Enter reply"
                style={
                  Object {
                    "width": "100%",
                  }
                }
                type="text"
              />
            </form>
          </div>
        </div>
        <ul
          className="list-group"
        />
      </div>
    </li>
    <li
      className="list-group-item"
      style={
        Object {
          "border": "1px solid #888",
          "borderRadius": 5,
          "boxShadow": "0px 5px 35px rgba(0, 0, 0, .7)",
          "marginBottom": "30px",
          "marginLeft": "30px",
          "marginRight": "30px",
          "padding": "0px 0px",
          "paddingBottom": "5px",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#fff",
            "padding": "10px",
          }
        }
      >
        <div
          className="container"
        >
          <div
            className="show-grid row"
          >
            <div
              className="col-md-8 col-xs-8"
            >
              <div>
                <p>
                  <span
                    aria-hidden={true}
                    className="fa fa-user"
                  />
                   
                  <b>
                    Arne
                  </b>
                </p>
                <p>
                     
                </p>
                <button
                  className="btn btn-default"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Like  
                  <span
                    aria-hidden={true}
                    className="fa fa-thumbs-up"
                  />
                </button>
                <button
                  className="btn btn-default"
                  disabled={false}
                  type="button"
                >
                  Reply  
                  <span
                    aria-hidden={true}
                    className="fa fa-reply"
                  />
                </button>
                <button
                  className="btn btn-default"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Delete  
                  <span
                    aria-hidden={true}
                    className="fa fa-remove-circle"
                  />
                </button>
              </div>
            </div>
            <div
              className="col-md-4 col-xs-4"
            >
              <p>
                <span
                  aria-hidden={true}
                  className="fa fa-clock-o"
                />
                 
                whatever
              </p>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "paddingLeft": "5px",
              "paddingRight": "5px",
              "width": "100%",
            }
          }
        >
          <button
            className="btn btn-primary"
            form="replyTo2"
            style={
              Object {
                "float": "right",
              }
            }
            type="submit"
            value="submitButton"
          >
            Submit Reply
          </button>
          <div
            style={
              Object {
                "overflow": "hidden",
                "paddingRight": "0.5rem",
              }
            }
          >
            <form
              className=""
              id="replyTo2"
              onSubmit={[Function]}
            >
              <input
                className="form-control"
                placeholder="Enter reply"
                style={
                  Object {
                    "width": "100%",
                  }
                }
                type="text"
              />
            </form>
          </div>
        </div>
        <ul
          className="list-group"
        />
      </div>
    </li>
    <li
      className="list-group-item"
      style={
        Object {
          "border": "1px solid #888",
          "borderRadius": 5,
          "boxShadow": "0px 5px 35px rgba(0, 0, 0, .7)",
          "marginBottom": "30px",
          "marginLeft": "30px",
          "marginRight": "30px",
          "padding": "0px 0px",
          "paddingBottom": "5px",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#fff",
            "padding": "10px",
          }
        }
      >
        <div
          className="container"
        >
          <div
            className="show-grid row"
          >
            <div
              className="col-md-8 col-xs-8"
            >
              <div>
                <p>
                  <span
                    aria-hidden={true}
                    className="fa fa-user"
                  />
                   
                  <b>
                    Kåre
                  </b>
                </p>
                <p>
                     
                </p>
                <button
                  className="btn btn-default"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Like  
                  <span
                    aria-hidden={true}
                    className="fa fa-thumbs-up"
                  />
                </button>
                <button
                  className="btn btn-default"
                  disabled={false}
                  type="button"
                >
                  Reply  
                  <span
                    aria-hidden={true}
                    className="fa fa-reply"
                  />
                </button>
                <button
                  className="btn btn-default"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Delete  
                  <span
                    aria-hidden={true}
                    className="fa fa-remove-circle"
                  />
                </button>
              </div>
            </div>
            <div
              className="col-md-4 col-xs-4"
            >
              <p>
                <span
                  aria-hidden={true}
                  className="fa fa-clock-o"
                />
                 
                whatever
              </p>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "paddingLeft": "5px",
              "paddingRight": "5px",
              "width": "100%",
            }
          }
        >
          <button
            className="btn btn-primary"
            form="replyTo3"
            style={
              Object {
                "float": "right",
              }
            }
            type="submit"
            value="submitButton"
          >
            Submit Reply
          </button>
          <div
            style={
              Object {
                "overflow": "hidden",
                "paddingRight": "0.5rem",
              }
            }
          >
            <form
              className=""
              id="replyTo3"
              onSubmit={[Function]}
            >
              <input
                className="form-control"
                placeholder="Enter reply"
                style={
                  Object {
                    "width": "100%",
                  }
                }
                type="text"
              />
            </form>
          </div>
        </div>
        <ul
          className="list-group"
        />
      </div>
    </li>
    <li
      className="list-group-item"
      style={
        Object {
          "border": "1px solid #888",
          "borderRadius": 5,
          "boxShadow": "0px 5px 35px rgba(0, 0, 0, .7)",
          "marginBottom": "30px",
          "marginLeft": "30px",
          "marginRight": "30px",
          "padding": "0px 0px",
          "paddingBottom": "5px",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#fff",
            "padding": "10px",
          }
        }
      >
        <div
          className="container"
        >
          <div
            className="show-grid row"
          >
            <div
              className="col-md-8 col-xs-8"
            >
              <div>
                <p>
                  <span
                    aria-hidden={true}
                    className="fa fa-user"
                  />
                   
                  <b>
                    Bjarne
                  </b>
                </p>
                <p>
                     
                </p>
                <button
                  className="btn btn-default"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Like  
                  <span
                    aria-hidden={true}
                    className="fa fa-thumbs-up"
                  />
                </button>
                <button
                  className="btn btn-default"
                  disabled={false}
                  type="button"
                >
                  Reply  
                  <span
                    aria-hidden={true}
                    className="fa fa-reply"
                  />
                </button>
                <button
                  className="btn btn-default"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Delete  
                  <span
                    aria-hidden={true}
                    className="fa fa-remove-circle"
                  />
                </button>
              </div>
            </div>
            <div
              className="col-md-4 col-xs-4"
            >
              <p>
                <span
                  aria-hidden={true}
                  className="fa fa-clock-o"
                />
                 
                whatever
              </p>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "paddingLeft": "5px",
              "paddingRight": "5px",
              "width": "100%",
            }
          }
        >
          <button
            className="btn btn-primary"
            form="replyTo4"
            style={
              Object {
                "float": "right",
              }
            }
            type="submit"
            value="submitButton"
          >
            Submit Reply
          </button>
          <div
            style={
              Object {
                "overflow": "hidden",
                "paddingRight": "0.5rem",
              }
            }
          >
            <form
              className=""
              id="replyTo4"
              onSubmit={[Function]}
            >
              <input
                className="form-control"
                placeholder="Enter reply"
                style={
                  Object {
                    "width": "100%",
                  }
                }
                type="text"
              />
            </form>
          </div>
        </div>
        <ul
          className="list-group"
        />
      </div>
    </li>
    <li
      className="list-group-item"
      style={
        Object {
          "border": "1px solid #888",
          "borderRadius": 5,
          "boxShadow": "0px 5px 35px rgba(0, 0, 0, .7)",
          "marginBottom": "30px",
          "marginLeft": "30px",
          "marginRight": "30px",
          "padding": "0px 0px",
          "paddingBottom": "5px",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#fff",
            "padding": "10px",
          }
        }
      >
        <div
          className="container"
        >
          <div
            className="show-grid row"
          >
            <div
              className="col-md-8 col-xs-8"
            >
              <div>
                <p>
                  <span
                    aria-hidden={true}
                    className="fa fa-user"
                  />
                   
                  <b>
                    Rolf
                  </b>
                </p>
                <p>
                     
                </p>
                <button
                  className="btn btn-default"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Like  
                  <span
                    aria-hidden={true}
                    className="fa fa-thumbs-up"
                  />
                </button>
                <button
                  className="btn btn-default"
                  disabled={false}
                  type="button"
                >
                  Reply  
                  <span
                    aria-hidden={true}
                    className="fa fa-reply"
                  />
                </button>
                <button
                  className="btn btn-default"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Delete  
                  <span
                    aria-hidden={true}
                    className="fa fa-remove-circle"
                  />
                </button>
              </div>
            </div>
            <div
              className="col-md-4 col-xs-4"
            >
              <p>
                <span
                  aria-hidden={true}
                  className="fa fa-clock-o"
                />
                 
                whatever
              </p>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "paddingLeft": "5px",
              "paddingRight": "5px",
              "width": "100%",
            }
          }
        >
          <button
            className="btn btn-primary"
            form="replyTo5"
            style={
              Object {
                "float": "right",
              }
            }
            type="submit"
            value="submitButton"
          >
            Submit Reply
          </button>
          <div
            style={
              Object {
                "overflow": "hidden",
                "paddingRight": "0.5rem",
              }
            }
          >
            <form
              className=""
              id="replyTo5"
              onSubmit={[Function]}
            >
              <input
                className="form-control"
                placeholder="Enter reply"
                style={
                  Object {
                    "width": "100%",
                  }
                }
                type="text"
              />
            </form>
          </div>
        </div>
        <ul
          className="list-group"
        />
      </div>
    </li>
  </ul>
  <div
    style={
      Object {
        "backgroundColor": "#fff",
        "bottom": "0px",
        "height": "50px",
        "paddingTop": "10px",
        "position": "fixed",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "marginLeft": "30px",
          "width": "85%",
        }
      }
    >
      <div
        style={
          Object {
            "paddingLeft": "5px",
            "paddingRight": "5px",
            "width": "100%",
          }
        }
      >
        <button
          className="btn btn-primary"
          form="submitQuestion"
          style={
            Object {
              "float": "right",
            }
          }
          type="submit"
          value="submitButton"
        >
          Submit Question
        </button>
        <div
          style={
            Object {
              "overflow": "hidden",
              "paddingRight": "0.5rem",
            }
          }
        >
          <form
            className=""
            id="submitQuestion"
            onSubmit={[Function]}
          >
            <input
              className="form-control"
              placeholder="Enter question"
              style={
                Object {
                  "width": "100%",
                }
              }
              type="text"
            />
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
